Gateway/Request/RefundDataBuilder.php

```diff
@@ -70,7 +70,7 @@ public function build(array $buildSubject)
         $pspReference = $payment->getCcTransId();
         $currency = $payment->getOrder()->getOrderCurrencyCode();
         $storeId = $order->getStoreId();
-        $merchantAccount = $this->adyenHelper->getAdyenAbstractConfigData("merchant_account", $storeId);
+        $merchantAccount = $this->adyenHelper->getAdyenMerchantAccount($payment, $storeId);
```

Comment:
You know send the whole payment object to the method where you only need:
$payment->getMethod() so just sent this into the call.